src/topology/instances/nnreal.lean

```diff
@@ -111,6 +111,14 @@ lemma summable_comp_injective {β : Type*} {f : α → nnreal} (hf : summable f)
 nnreal.summable_coe.1 $
 show summable ((coe ∘ f) ∘ i), from (nnreal.summable_coe.2 hf).comp_injective hi
 
+lemma summable_nat_add (f : ℕ → nnreal) (hf : summable f) (k : ℕ) : summable (λ i, f (i + k)) :=
```

Comment:
In `infinite_sum.lean`, you have `summable_nat_add_iff` that says something very similar (in fact a little bit stronger because there is an equivalence, which should also hold in `nnreal`), except it doesn't apply in your situation because `nnreal` is not an `add_comm_group`. Is it worth refactoring `summable_nat_add_iff` to let it apply in any `add_left_cancel_semigroup` with `has_continuous_add` (or whatever typeclass covers both situations uniformly)?

Reply:
I thought about this, but I'm honestly not sure whether the statement is true in an `add_left_cancel_semigroup`, because to state the underlying `has_sum` for the direction discussed here, you need some kind of subtraction. I thought about adding a type class for a sort of "weak subtraction" that is fulfilled by both `add_comm_group`s and things like `nnreal` and `nat`, but I came to the conclusion that it's not worth the effort.

However, maybe I'm missing something and the statement really is true in an `add_left_cancel_semigroup`. What do you think?

Reply:
Yes, I agree